File: Level 3/2xn 타일링 (dp) .cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
	int answer = 0;

	vector<int> dp(n + 3);

	dp[0] = 0;
	dp[1] = 1;
	dp[2] = 2;

	if (n > 3)
	{
		for (int i = 3;i <= n;i++)
		{
			dp[i] = (dp[i - 1] + dp[i - 2]) % 1000000007;
		}
	}

	answer = dp[n];

	return answer;
}

// https://programmers.co.kr/learn/courses/30/lessons/12900

File: Javascript/Level 3/2 x n 타일링 (dp) .js
function solution(n) {
    var answer = 0;
    var dp = [];

    dp[0] = 1;
    dp[1] = 2;

    if (n >= 2) {
        for (let i = 2; i < n; i++) {
            dp[i] = (dp[i - 1] + dp[i - 2]) % 1000000007;
        }
    }

    answer = dp[n - 1];

    return answer;
}

// https://programmers.co.kr/learn/courses/30/lessons/12900?language=javascript
// 1과 2를 조합해서 합이 n인 개수를 구하는 문제